src/components/common/DropDown/DropDownSelector.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useState, useRef, useMemo} from 'react';
+import {
+  TouchableOpacity,
+  FlatList,
+  ScrollView,
+  Keyboard,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import S from './DropDownSelector.style';
+
+type Props = {
+  value: string[];
+  options: string[];
+  onChange: (val: string) => void;
+  placeholder?: string;
+};
+
+const DropDownSelector = ({value, options, onChange, placeholder}: Props) => {
+  const [newOptionName, setNewOptionName] = useState<string>('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState<boolean>(false);
+  const inputRef = useRef(null);
+
+  const handleInputChange = (input: string) => {
+    const trimmedInput = input.trim();
+    setNewOptionName(trimmedInput);
+    setIsDropdownVisible(trimmedInput.length > 0);
+  };
+
+  const addNewOption = () => {
+    if (!newOptionName || options.includes(newOptionName)) return;
+    onChange(newOptionName);
+    setNewOptionName('');
+    setIsDropdownVisible(false);
+  };
+
+  const filteredOptions = useMemo(
+    () => options.filter(option => option.includes(newOptionName)),
+    [options, newOptionName],
+  );
+
+  const renderDropdownOption = ({item}: {item: string}) => (
+    <S.DropdownOption onPress={() => onChange(item)}>
+      <S.DropdownOptionName>{item}</S.DropdownOptionName>
+    </S.DropdownOption>
+  );
+
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+      <S.Container>
+        <S.SelectedItemsContainer>
+          <ScrollView horizontal>
+            {value.map((option, index) => (
+              <S.SelectedItem key={index}>
+                <S.SelectedItemName>{option}</S.SelectedItemName>
+                <TouchableOpacity onPress={() => onChange(option)}>
```

Comment:
```suggestion
            {value.map((option) => (
              <S.SelectedItem key={option}>
                <S.SelectedItemName>{option}</S.SelectedItemName>
                <TouchableOpacity onPress={() => onChange(option)}>
```

value의 각 요소들이 유니크하니 key로 index 대신 option을 사용해도 되지 않을까 싶어요

Reply:
해당 부분 props 구조 변경으로 인해 option.id 사용으로 변경했습니다~